run_shellcheck.py

```diff
@@ -0,0 +1,26 @@
+# This script serves as a wrapper around ShellCheck to run ShellCheck only
+# files listed in the .shellcheckignore script since ShellCheck does not support
+# this functionality by default
+
+from pathlib import Path
+import subprocess
+
+ignored_files = set()
+with open('.shellcheckignore', 'r') as ignore_file:
+    for line in ignore_file.readlines():
+        ignored_files.update(Path('.').rglob(line.strip()))
+
+all_shell_scripts = set(Path('.').rglob('*.sh'))
+
+shell_scripts_to_check = [str(x) for x in all_shell_scripts.difference(ignored_files)]
+
+print(['shellcheck'] + shell_scripts_to_check)
+
+return_code = 0
+for script in shell_scripts_to_check:
+    process = subprocess.run(['shellcheck', script], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
```suggestion
    process = subprocess.run(['shellcheck', '-Calways', script], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
```

Maybe also print out standard error here?  Also consider adding color output?

Reply:
When I tried adding STDERR, ShellCheck only printed newlines to stderr for both successful and unsuccessful runs.  I'll see if I can add it such that the interleaving with STDOUT remains sensible.  I also tried adding color output but I was unable to get it to work after several hours of work on it.  If you have suggestions, I'm definitely open to trying them.

Reply:
Adding `-Calways` as suggested works for me in getting it to have color. I don't think there's any way to do it otherwise.

> When I tried adding STDERR, ShellCheck only printed newlines to stderr for both successful and unsuccessful runs. I'll see if I can add it such that the interleaving with STDOUT remains sensible. 

It looks like shellcheck only outputs to stdout in all cases, and so I think makes sense to just ignore stderr completely.

We should wrap `process.stdout.print` in a check that it's not empty as well so that we don't get an empty line per checked file that's good.